### DISCUSSION TITLE
[Question] How to get all item definitions of app?

### DISCUSSION THREAD

**Discussion by iocmet** [SteamRE/SteamKit#1253] [General]
I almost not know anything about steam web api, is it possible? I need dump it to json

**Comment by SeRi0uS007** [comment]
According to my experience getting item definitions is different for Valve and non-Valve games.

For non-Valve games the procedure is as follows:

1. Get the digest: [GetItemDefMeta](https://steamapi.xpaw.me/#IInventoryService/GetItemDefMeta)
2. Using the digest, get a complete database of all item definitions for a specific app id: [GetItemDefArchive](https://steamapi.xpaw.me/#IGameInventory/GetItemDefArchive)

Note: at some point something might go wrong and the response from the Steam server either for the first or second point will be null or any other errors.

Regarding Valve games:
Each game has its own characteristics but basically, [SteamTracking](https://github.com/SteamDatabase/SteamTracking) contains a file with all item definitions.

**Comment by iocmet** [comment]
Thanks

**Comment by iocmet** [comment]
It works
